spec/components/schemas/message/all.ts

```diff
@@ -24,8 +24,8 @@ const all: SchemaObject = {
         minItems: 1,
       },
       messageIdRef: {
-        title: 'Message Reply ID',
-        description: 'Replied message unique indentifier',
+        title: 'read only',
+        description: 'When an user sends a message quoting a previous message, the idetifier of the quoted message will be provided here.<br>*Only applicable to [WhatsApp channel](#tag/WhatsApp).*',
```

Comment:
Faltou só o atributo readOnly: true

Reply:
Qual é a diferença, visualmente falando?

Reply:
Atualmente, nenhuma, mas o redoc pode vir a ser atualizado e fazer diferença.

Mas faz diferença pra outras ferramentas (como o validador que usamos no api-control) e ferramentas que geram client automaticamente, então mesmo não fazendo diferença na documentação, é importante!

Reply:
E prestando atenção melhor, esse campo deveria estar no base.ts igual aos outros.

Reply:
Tem algum motivo especial pra ter sido colocado aqui?

Reply:
Beleza, incluir é de boa, só achei que teria alguma diferença visual.

Reply:
Não coloquei no base.ts, pois os atributos daí também são referenciados pelo status message, e esse atributo é exclusivo de message.

Reply:
Não no base.ts do messages.

Reply:
https://github.com/zenvia/zenvia-openapi-spec/blob/master/spec/components/schemas/message/base.ts

Reply:
Entendi que isso era exclusivo de message de webhook, não é isso?

Reply:
É sim, é o que significa readOnly! Mas como você pode ver nesse base.ts, tem vários readOnly lá!

Reply:
Tudo bem que não é retornado igual o visitor.

Reply:
pode ser melhor deixar só aqui mesmo.

Reply:
Ah, beleza então! Sim, tem mesmo, até achei que tinha alguma noção de visualização mesmo, quando comentou... Eu já mandei a atualização incluindo o atributo.

Reply:
nesse caso o readOnly nem faz tanta diferença, mas é bom mesmo assim, caso o redoc venha a ter algo visual pra isso.

Reply:
Se colocar como read-only ele não vai ser exibido no post, então acho que faz diferença sim. O title deve manter o titulo do campo e não "read only", tem outra chave para isso.

Reply:
@hmagarotto , quanto ao read-only o campo é (e a descrição de read-only), a chave pra isso é esse mesma, o problema é que o redoc não mostra nada de diferente. E readOnly, e ele é renderizado sim, e por padrão (ao contrário do writeOnly. Na parte de webhook você consegue ver, igual a outros campos comuns: channel, direction, id: https://github.com/zenvia/zenvia-openapi-spec/blob/master/spec/components/schemas/message/base.ts

Quanto ao title, sim, é uma distorção dele intencional que estou propondo fazer, pois o hoje o redoc não tem nenhum indicador se o campo é readOnly ou writeOnly, o que fica bem confuso. E o legal é que esse valor é rederizado bem no tipo do campo, e que não agrega muito spó pra duplicar o nome do campo com espaço que tem sido o uso atual dele. Obviamente, poderia ser feito de algum outro jeito isso (talvez via CSS ou um plugin no redoc), mas realmente precisamos de um jeito de deixar claro isso.

Um bom exemplo de como isso está confuso é o campo payload do conteúdo text: geral que eu vejo testar a API a primeira vez só olhando a documentação preenche ele. Inclusive @hmagarotto precisamos ver se mantemos ele ou corrigimos ele, pois hoje ele não é nunca preenchido no api-control, só tá ali pra confundir o post.

Reply:
Mas lembrando que esse campo novo tá no all.ts e só aparece no webhook, não faz diferença esse read-only no title, então realmente é melhor voltar pro valor anterior pra simplificar @joaoneves

Reply:
@stefanymiguel10 o que você achou da minha sugestão de frase? Tá boa?

Reply:
Está, sim @rafael-org. ;)

Reply:
Ficou:
When an user sends a message quoting a previous message, the idetifier of the quoted message will be provided here.<br>*Only applicable to [WhatsApp channel](#tag/WhatsApp).*

Reply:
Legal. @joaoneves olhandoo agora, a palavra "identifier" tá errado (tá "idetifier"). Tem as manhas de corrigir?

Reply:
@rafael-org  O Campo payload não é exibido no POST, justamente por ser read-only.
![Screenshot_20201211_195031](https://user-images.githubusercontent.com/3268876/101962176-36a37480-3bea-11eb-9eea-ca6d87761796.png)

Reply:
@hmagarotto Na verdade mesmo que ele não fosse readOnly ele não apareceria no post porque ele tá no arquivo all.ts, que é usado só no webhook, e que inclusive foi intencional conforme o João explicou (não tá sendo implementada a opção de fazer referencia a uma mensagem na postagem, só de receber).

Reply:
e aí o readOnly é indeferente na real.

Reply:
Sim, neste caso. Mas se tivesse na base iria funcionar o read-only sem problemas.

Reply:
SIm, mas falei pra por pela mesma premissa: de que era só no webhook pois tinha também entendido que não haveria opção de referenciar mensagens no envio. Se fosse ter, definitivamente estaria errado kkkkk